### PR TITLE
sly-5-6

### PR DESCRIPTION
**Pull request by shyam-dev10**
review the backend authentication API's

### CODE REVIEW

File: server/controller/Auth.js [hunk 1]
```diff
@@ -1,94 +1,91 @@
...
-      if (!emailRegex.test(email)) {
-        return res.status(400).json({ error: 'Please provide a valid email address' });
-      }
-
-      // Check if the user exists
       const user = await User.findOne({ email });
-
+  
       if (!user) {
         return res.status(401).json({ error: 'Invalid credentials' });
```
Comment (pavankpdev) [inline server/controller/Auth.js]: Incorrect error message

File: server/index.js [hunk 2]
```diff
@@ -1,36 +1,18 @@
...
-import userRoutes from './routes/userRoutes.js'
-import Product from './models/Product.js';
-
-const port = process.env.PORT || 4000;
-
+import dotenv from 'dotenv'
+import routes from './routes/routes.js'
+import db from './database/connection.js';
+dotenv.config();
 connectDB();
```
Comment (pavankpdev) [inline server/index.js]: Where is this function declared? There's no imports specified? Was this code tested?

File: server/routes/routes.js
Comment (pavankpdev) [inline server/routes/routes.js:1]: Invalid file name

File: server/controller/Auth.js [hunk 3]
```diff
@@ -1,94 +1,91 @@
...
       res.status(200).json({ token });
     } catch (error) {
       console.error(error);
       res.status(500).json({ error: 'Error logging in' });
     }
-  }
-}
+  };
+
+  const logoutUser = (req, res) => {
```
Comment (pavankpdev) [inline server/controller/Auth.js]: No API necessary for logout

File: server/index.js (hunk 2 above)
Comment (pavankpdev) [inline server/index.js]: This returns a Promise, which is not handled

Comment (pavankpdev) [reply server/index.js]: This will crash the server, if in case the DB fails to connect, with the error `unhandled promise rejection`

**Review by pavankpdev — CHANGES_REQUESTED** [review]
Code looks good, but seems like this PR was created without proper testing, the `index.js` file is trying to call `connectDB()` with no imports or declaration of the function.

File: server/controller/Auth.js (hunk 3 above)
Comment (shyam-dev10) [reply server/controller/Auth.js]: Can i remove the logout function then??

Comment (pavankpdev) [reply server/controller/Auth.js]: yeah

File: server/index.js (hunk 2 above)
Comment (pavankpdev) [reply server/index.js]: I don't how this is resolved? What is the `connectDB()` imported here?

Comment (pavankpdev) [reply server/index.js]: The promise is still not handled, here's a hint use `try` `catch` block

File: server/routes/routes.js
Comment (pavankpdev) [reply server/routes/routes.js:1]: The file name is still `route.js` it should be `auth.js`

File: server/middleware/authm.js
Comment (pavankpdev) [inline server/middleware/authm.js:1]: Why do you have auth logic in middleware, this is invalid, middleware should be a part of the auth logic, not the entire logic, refactor this

**Review by pavankpdev — CHANGES_REQUESTED** [review]

File: server/routes/routes.js
Comment (shyam-dev10) [reply server/routes/routes.js:1]: changed it!

File: server/controller/Auth.js (hunk 3 above)
Comment (shyam-dev10) [reply server/controller/Auth.js]: removed logout api

File: server/index.js (hunk 2 above)
Comment (shyam-dev10) [reply server/index.js]: used try and catch to handle it

File: server/middleware/authm.js
Comment (shyam-dev10) [reply server/middleware/authm.js:1]: my bad. refactored this

File: server/controller/Auth.js (hunk 3 above)
Comment (pavankpdev) [reply server/controller/Auth.js]: I still see the logout function here, why?

File: server/index.js [hunk 4]
```diff
@@ -1,36 +1,18 @@
...
+import routes from './routes/auth.js'
+import db from './database/db.js';
 
-const port = process.env.PORT || 4000;
-
-connectDB();
+dotenv.config();
 const app = express();
+const PORT = process.env.PORT || 4000;
+db();
```
Comment (pavankpdev) [inline server/index.js]: Reported this multiple times already, `db()` returns `Promise` which should be handled. I recommend learning about JavaScript Promises, before fixing this

**Review by pavankpdev — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 4 above)
Comment (shyam-dev10) [reply server/index.js]: ok i will learn about it and ill try to fix it

Comment (Shyam-crypto) [reply server/index.js:9]: can i do like this
async function startApp(){
try{
await db();
   }
console.log('');
}catch(error){
console.error('',error.message);
}
}
startApp();

File: server/index.js [hunk 5]
```diff
@@ -6,13 +6,23 @@ import db from './database/db.js';
...
 app.use(express.json());
 app.use('/api/auth', routes);
 
-try {
-    app.listen(PORT, () => console.log(`Server started on port ${PORT}`));
-    } catch (error) {
-        console.error("connection failed");
-    }
\ No newline at end of file
+ async function startApp(){
```
Comment (pavankpdev) [inline server/index.js:13]: This is valid, but it's just redundant code, try this

```javascript
db()
   .then(() => {
       console.log("DB: running")
       app.listen(PORT, () => {
            console.log(`Server started on port ${PORT}`)
        });
   })
   .catch(console.log)
```

Comment (pavankpdev) [reply server/index.js:13]: This way, you'll handle the error, and also start the server only if DB connection is successful, also less code

**Review by pavankpdev — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 5 above)
Comment (shyam-dev10) [reply server/index.js:13]: Okay

**Review by pavankpdev — APPROVED** [review]
Looks good! merging!

**Pull request merged**